src/zeal/docset.py

```diff
@@ -1,28 +1,30 @@
 import os
+import platform
 import shutil
+import subprocess
 import urllib
+from pathlib import Path
 
 import bs4
 
 from . import config, downloads, exceptions
 
 LATEST_VERSION = "latest"
 
-def list_all(docset_dir: str = config.docset_dir) -> list:
+def list_all(docset_dir: Path = config.docset_dir) -> list:
     """List the docsets in the docset_dir.
 
     :param docset_dir: String, path to the Zeal docset directory. DefaultL filesystem.docset_dir
     :return: List of docsets (by name, not by path)
     """
-    files_list = os.listdir(docset_dir)
     installed_docsets = []
-    for file in files_list:
-        if os.path.isdir(os.path.join(docset_dir, file)) and file.endswith(".docset"):
-            installed_docsets.append(file.removesuffix(".docset"))
+    for elem in docset_dir.glob("*.docset"):
```

Comment:
Be more specific than `elem` with the variable name.

Reply:
as glob actually returns iterator of Path objects we can call them just `path`? Previous `file` is invalid because of global re-definition of function `file` and because returned paths are not only files (could be also folders and links).